learn/functions/lambda.md

```diff
@@ -3,32 +3,49 @@
 
 **Lambda functions** in Python are **anonymous** functions that can be defined and declared **inline** with the rest of the code. 
 
-## What is a lambda function in Python
+## What is a Lambda Function in Python
 
-Lambda functions take in parameters and return a value like a regular Python function, but are **usually only used once** and therefore not given a name. Instead, they are **passed as an argument to other functions** or used in an expression.
+In Python, a lambda function is a concise way to create small, anonymous functions. Unlike regular functions defined with the `def` keyword, lambda functions are created using the `lambda` keyword and are typically used for short, immediate operations.
 
-### Example 1: Defining a lambda function to multiply two numbers
+The syntax for a lambda function is simple:
 
-```python 
-multiply = lambda x, y: x * y
-
-# calling the lambda function
-print(multiply(5, 3)) # output: 15
+```python
+lambda arguments: expression
 ```
 
-### Example 2: Using a lambda function as an argument to the map() function
+Here, `lambda` is the keyword, `arguments` are the function's input parameters, and `expression` is the operation the function performs.
 
+Here are also some examples of declaration and usage:
+
+1. **Basic Addition**
 ```python
-my_list = [1, 2, 3, 4, 5]
-squared_list = list(map(lambda x: x**2, my_list))
+add = lambda x, y: x + y
+print(add(3, 4))  # Output: 7
+```
 
-# printing the squared list
-print(squared_list) # output: [1, 4, 9, 16, 25]
+2. **Using Within Higher-Order Functions**
+```python
+numbers = [1, 2, 3, 4, 5]
+doubled = list(map(lambda x: x * 2, numbers))
+print(doubled)  # Output: [2, 4, 6, 8, 10]
+```
+
+3. **Sorting a List of Tuples**
+```python
+students = [('Alice', 25), ('Bob', 20), ('Charlie', 30)]
+students.sort(key=lambda x: x[1])
+print(students)  # Output: [('Bob', 20), ('Alice', 25), ('Charlie', 30)]
 ```
 
-In the first example, a `lambda` function `multiply` is defined that **takes in two parameters** and returns their product. In the second example, the lambda function **is used as an argument** to the `map()` function to square each element in the list. Lambda functions are often used in conjunction with other functions **to make code more concise and readable.**
+Lambda functions are particularly handy when a small function is needed for a brief period. Their simplicity is beneficial in cases where defining a regular function might be overkill.
+
+Key Points to Remember:
```

Comment:
If it's a heading - highlight it. If it's just a sentence - don't use Camel Case.

Reply:
corrected